src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,721 @@
+                ref={inputRef}
+                value={search}
+                onChange={onUpdateSearch}
+                onKeyDown={onKeyDown}
+                placeholder={placeholder}
+              />
+              <SelectList>
+                <SelectListContent>
+                  <OptionContext.Provider value={optionContextValue}>
+                    {options && options.length > 0 ? (
+                      <AutoCompleteOptionList
+                        options={options}
+                        id={id ?? defaultId}
+                      />
+                    ) : (
+                      children
+                    )}
+                  </OptionContext.Provider>
+                </SelectListContent>
+                {visibleList.current.length === 0 && (
+                  <SelectNoDataContainer {...props}>
+                    No Options found{search.length > 0 ? ` for "${search}" ` : ""}
+                  </SelectNoDataContainer>
+                )}
+              </SelectList>
+            </PopoverContent>
+          </Portal>
+        </SelectPopoverRoot>
+        {!!error && error !== true && <Error>{error}</Error>}
+      </FormElementContainer>
+      {label && (
+        <Label
+          htmlFor={id ?? defaultId}
+          disabled={disabled}
+          error={!!error}
+        >
+          {label}
+        </Label>
+      )}
+    </FormRoot>
+  );
+};
+
+export const Group = forwardRef<HTMLDivElement, SelectGroupProps>(
+  ({ children, heading, ...props }, forwardedRef) => {
+    useSearch();
+    return (
+      <SelectGroupContainer
+        {...props}
+        ref={mergeRefs([
+          forwardedRef,
+          node => {
+            const hidden = node?.querySelectorAll("[cui-autocomplete-item]").length === 0;
+            if (hidden) {
+              node?.setAttribute("hidden", "");
+            } else {
+              node?.removeAttribute("hidden");
+            }
+            node?.setAttribute("aria-hidden", hidden.toString());
+          },
+        ])}
+      >
+        <SelectGroupName>{heading}</SelectGroupName>
+        <SelectGroupContent>{children}</SelectGroupContent>
+      </SelectGroupContainer>
+    );
+  }
+);
+
+Group.displayName = "AutoComplete.Group";
+
+const CheckIcon = styled.svg<{ $showCheck: boolean }>`
+  opacity: ${({ $showCheck }) => ($showCheck ? 1 : 0)};
+`;
+
+export const Item = forwardRef<HTMLDivElement, SelectItemProps>(
+  (
+    {
+      disabled = false,
+      children,
+      label,
+      separator,
+      onSelect: onSelectProp,
+      value = "",
+      icon,
+      iconDir,
+      onMouseOver: onMouseOverProp,
+      ...props
+    },
+    forwardedRef
+  ) => {
+    const { highlighted, updateHighlighted, isHidden, selectedValue, onSelect } =
+      useOption();
+    const onSelectValue = () => {
+      if (!disabled) {
+        onSelect(value);
+        if (typeof onSelectProp == "function") {
+          onSelectProp(value);
+        }
+      }
+    };
```

Comment:
when a new option is selected we should update the search value as well

Reply:
@serdec No this is indented the search value should not be selected as the element whould revert to its original state
If the user clicks again and the search should start from the begining